klp_tc_functions.sh

```diff
@@ -311,6 +312,14 @@ if [ ! -f $KLP_ENV_CACHE_FILE ]; then
         echo "0" >> $KLP_ENV_CACHE_FILE
     fi
 
+    echo -n 'export KLP_TEST_REG_API=' >> $KLP_ENV_CACHE_FILE
+    if grep klp_register_patch "/lib/modules/$(uname -r)/source/include/linux/livepatch.h"
```

Comment:
The kernel header file is yet another loose part in the system. Of course, without it the test modules won't compile, but I'd still prefer making the decision based on information provided by running kernel image. I suggest either grepping `/proc/kallsyms` or checking kernel version. Cf. https://github.com/lpechacek/qa_test_klp/blob/19692b48765a936996f1e7775700c62de1c4bfeb/klp_tc_functions.sh#L336-L339

Reply:
Sure :). I found this commit [958ef1e39d24 livepatch: Simplify API by removing registration step](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=958ef1e39d24d6cb8bf2a7406130a98c9564230f) merged into v5.1-rc1, so I'll `$VERSION_CODE` for it.

Reply:
Uh, took me a while to realize that you offered code to copy paste :)

Reply:
@lpechacek Libor, this will not work for SLE12 SP4 (old version, but backported patches). I'll also add check about `KLP_NOREG_API` not being defined. But IMHO grepping headers is more reliable for all distros.